build_stage_repository.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python3
+
+from subprocess import check_output, STDOUT, CalledProcessError
+import os
+import yaml
+import glob
+import shutil
+import time
+import hashlib
+import sys
+
+
+def execute(command):
+    try:
+        return check_output(command, universal_newlines=True, stderr=STDOUT)
+    except CalledProcessError as error:
+        print(error.output)
+        return False
+
+
+def move_rpms_from_copr_to_stage(collection, version, src_folder, dest_folder, source=False):
+    if source:
+        print(f"Moving {collection} Source RPMs from Copr directory to stage repository")
+    else:
+        print(f"Moving {collection} RPMs from Copr directory to stage repository")
+
+    if not os.path.exists(dest_folder):
+        os.mkdir(dest_folder)
+
+    repo_folder = f"{src_folder}/el8-{collection}-{version}"
+
+    if source:
+        files = glob.glob(repo_folder + f"/**/*.src.rpm")
+    else:
+        files = glob.glob(repo_folder + f"/**/*.rpm")
+
+    for file in files:
+        file_name = os.path.basename(file)
+        shutil.move(file, os.path.join(dest_folder, file_name))
+
+    shutil.rmtree(src_folder)
+
+def modulemd_yaml(collection):
+    return f"modulemd/modulemd-{collection}-el8.yaml"
+
+
+def generate_modulemd_version(version):
+    if version == 'nightly':
+        modulemd_version_prefix = '9999'
+    else:
+        major, minor = version.split('.')
+        modulemd_version_prefix = int(major)*100 + int(minor)
+
+    modulemd_version_string = time.strftime(f"{modulemd_version_prefix}%Y%m%d%H%M%S", time.gmtime())
+
+    return int(modulemd_version_string)
+
+
+def generate_modulemd_context(collection, version):
+    context_string = f"{collection}-{version}"
+    digest = hashlib.sha256(context_string.encode()).hexdigest()
+    return digest[:8]
+
+
+def create_modulemd(collection, version, stage_dir):
+    print("Adding modulemd to stage repository")
+    cmd = [
+        'rpm',
+        '--nosignature',
+        '--query',
+        '--package',
+        f"{stage_dir}/*rpm",
+        "--queryformat=%{name}-%{epochnum}:%{version}-%{release}.%{arch}\n"
+    ]
+    output = execute(cmd)
+
+    with open(modulemd_yaml(collection), 'r') as file:
+        modules = yaml.safe_load(file)
+
+    modules['data']['artifacts'] = {'rpms': output.splitlines()}
```

Comment:
I did some type checking and `output` can be `False` and `False.splitlines()` will fail.